src/flag.rs

```diff
@@ -164,8 +176,12 @@ mod tests {
 
     fn self_signal() {
         unsafe {
+            #[cfg(not(windows))]
             let pid = libc::getpid();
+            #[cfg(not(windows))]
             libc::kill(pid, ::SIGUSR1);
+            #[cfg(windows)]
+            libc::raise(::SIGTERM);
```

Comment:
Here I actually prefer the way you wrote it (with the windows/not-windows conditions), because the getpid/kill combination allows delivering the signal to arbitrary thread, not just the current one. This allows some more bugs to manifest.

But a comment giving the rationale would be nice (eg. explaining the advantage of getpid/kill, vs Windows not having kill)

Reply:
I actually didn't know the difference between `raise` and `kill(getpid)`. I'll unify them as you said.